client/src/review-components/ReviewForm.jsx

```diff
@@ -1,76 +1,145 @@
-import React from 'react';
+import React, {useState} from 'react';
 import ReactDOM from 'react-dom/client';
-import Stars from './Stars.jsx';
+import HoverStars from './HoverStars.jsx';
+import axios from 'axios';
 
 function ReviewForm(props) {
+
+  let charDetails = {
+    Size: ['A size too small', '½ a size too small', 'Perfect', '½ a size too big', 'A size too wide'],
+    Width: ['Too narrow', 'Slightly narrow', 'Perfect', 'Slightly wide', 'Too wide'],
+    Comfort: ['Uncomfortable', 'Slightly uncomfortable', 'Ok', 'Comfortable', 'Perfect'],
+    Quality: ['Poor', 'Below average', 'What I expected', 'Pretty great', 'Perfect'],
+    Length: ['Runs short', 'Runs slightly short', 'Perfect', 'Runs slightly long', 'Runs long'],
+    Fit: ['Runs tight', 'Runs slightly tight', 'Perfect', 'Runs slightly long', 'Runs loose']
+  }
+
+  let charArray = Object.keys(props.data.characteristics).map((element, index) => {
```

Comment:
i suggest optional chaining to protect your widget from failing due to missing data

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining